plutus-pab/src/Plutus/PAB/Core/ContractInstance/BlockchainEnv.hs

```diff
@@ -97,24 +96,19 @@ blockAndSlot BlockchainEnv{beCurrentBlock, beCurrentSlot} =
 -- | Process a chain sync event that we receive from the alonzo node client
 processChainSyncEvent :: BlockchainEnv -> ChainSyncEvent -> STM (Either SyncActionFailure (Slot, BlockNumber))
 processChainSyncEvent blockchainEnv event = do
-  result <- case event of
-              Resume _ -> Right <$> blockAndSlot blockchainEnv
-              RollForward (BlockInMode (C.Block header transactions) era) _ ->
-                case era of
-                  -- Unfortunately, we need to pattern match again all eras because
-                  -- 'processBlock' has the constraints 'C.IsCardanoEra era', but not
-                  -- 'C.BlockInMode'.
-                  C.ByronEraInCardanoMode   -> processBlock header blockchainEnv transactions era
-                  C.ShelleyEraInCardanoMode -> processBlock header blockchainEnv transactions era
-                  C.AllegraEraInCardanoMode -> processBlock header blockchainEnv transactions era
-                  C.MaryEraInCardanoMode    -> processBlock header blockchainEnv transactions era
-                  C.AlonzoEraInCardanoMode  -> processBlock header blockchainEnv transactions era
-              RollBackward chainPoint _ -> runRollback blockchainEnv chainPoint
-  flip traverse_ result $ \(_, BlockNumber n) -> do
-    if n `mod` 5_000 == 0
-      then garbageCollect blockchainEnv
-      else pure ()
-  pure result
+  case event of
+    Resume _ -> Right <$> blockAndSlot blockchainEnv
+    RollForward (BlockInMode (C.Block header transactions) era) _ ->
+      case era of
+        -- Unfortunately, we need to pattern match again all eras because
+        -- 'processBlock' has the constraints 'C.IsCardanoEra era', but not
+        -- 'C.BlockInMode'.
+        C.ByronEraInCardanoMode   -> processBlock header blockchainEnv transactions era
```

Comment:
I know this is older code, but isn't the simple solution here to let `processBlock` take the whole `BlockInMode` value?

Reply:
I think this is written this way because you need a 'proof' that `C.IsCardanoEra era` (required by a constraint on `processBlock`) and that is provided by pattern matching on the era part. I may not understand what you are asking though.